src/main/java/banking/database/AccountDao.java

```diff
@@ -0,0 +1,101 @@
+package banking.database;
+
+import banking.model.Account;
+import banking.model.Card;
+
+import java.sql.*;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+
+public class AccountDao {
+
+    private final DBManager dbManager;
+
+    public AccountDao(DBManager dbManager) {
+        this.dbManager = dbManager;
+    }
+
+    public Optional<Account> get(String cardNumber, String cardPIN) {
+        try (Connection connection = dbManager.createConnection();
```

Comment:
Opening a new connections each time is a bad idea. Please research how we can improve it

Reply:
Created field that stores Connection object and added methods to get and close connection when needed. So now connection is created only one time

Reply:
Can you also explain some best practices for managing DB connections? Or maybe conduct a workshop about JDBC?

Reply:
I guess you are leading me to the ConnectionPool pattern, right? I have just read some topics about that and can say that will definitely go over the details of how it works and implement it in this project